flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1255,6 +1255,21 @@ private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint)
             } else {
                 lastSubsumed = null;
             }
+
+            PendingCheckpointStats statsCallback = getStatsCallback(pendingCheckpoint);
+            if (statsCallback != null) {
+                LOG.trace(
+                        "Checkpoint {} size: {}Kb, duration: {}ms",
+                        checkpointId,
+                        statsCallback.getStateSize() == 0 ? 0 : statsCallback.getStateSize() / 1024,
+                        statsCallback.getEndToEndDuration());
+                // Finalize the statsCallback and give the completed checkpoint a
+                // callback for discards.
+                CompletedCheckpointStats.DiscardCallback discardCallback =
+                        statsCallback.reportCompletedCheckpoint(
+                                completedCheckpoint.getExternalPointer());
+                completedCheckpoint.setDiscardCallback(discardCallback);
+            }
```

Comment:
Can you extract some of the code from this method to shrink it and increase readability? Maybe extract this new code to a new method and the lines L1264:L1282 to another new method (`cleanupAfterCompletedCheckpoint`?)